app.js

```diff
@@ -112,6 +112,36 @@ function roundCounter(){
   roundCounter.innerText = currentRound;
 }
 
+// function setItems()
+function setItems(key, value) {
+  let stringified = JSON.stringify(value);
+  localStorage.setItem(key, stringified);
+}
+
+// function getItems()
+function getItems(key) {
+  let stringified =localStorage.getItem(key);
+  let parsed = JSON.parse(stringified);
+  return parsed;
+}
+
+//get updated product array and round from local storage if they exist, reset survey after 25 rounds
+function onLoad(){
+  if (getItems('allProducts')) {
+    allProducts = getItems('allProducts');
+  }
+
+  if (getItems('round')) {
+    currentRound = getItems('round');
```

Comment:
Since from previous labs we're disabling event listeners at 25 rounds, setting your currentRound variable to whatever was stored is counterproductive to our goal if a user completes 25 rounds in the previous visit. If you want to track the total number of rounds with your counter still, I would suggest setting your stored data to a separate variable and adding them wherever you want to display the rounds.